src/Streamly/Internal/Data/Parser.hs

```diff
@@ -543,14 +545,16 @@ sliceSepByP _cond = undefined -- D.toParserK . D.sliceSepByP cond
 sliceSepBy :: MonadCatch m => (a -> Bool) -> Fold m a b -> Parser m a b
 sliceSepBy cond = D.toParserK . D.sliceSepBy cond
 
--- | Like 'sliceSepBy' but does not drop the separator element, instead
+-- | takes until a seperator is found, gives it back to the input
```

Comment:
This doc needs to be more clear. I think this statement `takes until a seperator is found, gives it back to the input if the seperator is not the first element, else if it is the first element, then we only take the seperator and stop.` is equivalent to `if x then y else y` and hence hot required.
Also, Change `seperator` to `separator`